<html>
<head>
<!-- Latest compiled and minified CSS -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

<!-- Optional theme -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">

<!-- Latest compiled and minified JavaScript -->
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
</head>

<style>
body { margin-left: 20px }
</style>
<body>
<h2><a href="../">CRISPOR</a> Changes</h2>
<p>Many thanks to the user that have contacted me, suggested features and tested them!</p>

<h4>Version 4.3, May 2017</h4>
<ul>
<li>Saturation Mutagenesis Assistant (Matt Canver, Luca Pinello)
<li>Genbank-like formats for various plasmid editors and Fasta export (Jean-Paul)
<li>Better command line support (Luca Pinello)
<li>PCR page: Fixed bad bug in T7 plasmid primer (many thanks to Macie Walker)
<li>PCR page: Changed lentiCrisprV2 enzyme Bbs1 -> BsmBI (thanks to Erin Simonds, UCSF)
<li>Fixed crash with unicode in batch name (thanks to Thom Saunders, U Mich)
<li>External websites can now pre-fill out the input form with values for the genome (parameter: "org") or a sequence (parameter: "seq"). Thanks to Patrick Lemaire, U Montpellier, and Frank Perez, INSERM Paris
</ul>

<h4>Version 4.2, Apr 2017</h4>
<ul>
<li>The whole site is around 10x faster now.  Mark Diekhans/UCSC fixed the twoBit tool I use, many thanks!
<li>Genome Variants can be overlaid onto the input sequence (Matthew Canver/Harvard, Jean-Paul Concordet)
<li>Saturation Mutagenesis Support (Matt Canver, Luca Pinello)
<li>Genbank-like formats for various plasmid editors and Fasta export (Jean-Paul)
<li>Better command line support (Luca Pinello)
<li>Support for the Cpf1 enzyme. Its PAM is TTTN. (Max)
<li>Input: Guides cannot overlap Ns in the sequence (Alberto Stolfi, NYU)
<li>Input: A name for the input sequence can now be shown on all pages (Jean-Paul)
<li>Main Page: Added many more restriction enzymes and highlighted on PCR product (Saulius Zuklys, Uni Basel)
<li>Main Page: Guides can be sorted by position (Virginie Virolle, Universite Nice)
<li>Main Page: Click OOF score to show Bae et al. microhomology (Tsuyoshi Momose, Villefranche)
<li>Main Page: Primers can be designed for off-targets and results exported to Crispresso (Max and Luca Pinello/Harvard)
<li>Main Page: Showing only the best two scores now, based on our paper. Others are optional. (Max) 
<li>Main Page: Sorting the off-targets by CFD score now (Max, Virginie Virolle)
<li>Main Page: Show only guides that start with G- or A- (Peter Rogowsky, ENS Lyon)
<li>Download: Adding sequence and version info to Excel file (Soren Warming/Max)
<li>PCR page: Fixed bad bug in T7 plasmid primer (many thanks to Macie Walker)
<li>PCR page: Dropdown to select the Addgene plasmid, changes the cloning sites (Jean-Paul Concordet)
<li>PCR Page: Tm and target length can be selected for the validation primers (Jean-Michel Heslan, Univ Nantes)
<li>PCR Page: Redesign of the input validation primers page (Max)
<li>PCR Page - more protocols: cloning in Ciona (Alberto), off-target sequencing (Matthew Canver, MIT), synthesis (Ann de Cian, MNHN Paris)
<li>UCSC Browser: Browser tracks with pre-computed guides for the major model organisms (Max)
<li>Local install: Off-target Excel export now includes the strand (Max). This changes the output format but was necessary.
<li>Local installation/offline-mode: all .js and .css files are now included in the distribution
<li>Local install: license is now free for academic use only. (Max)
</ul>

</body>
</html
